src/main/java/org/folio/rest/impl/BLUsersAPI.java

```diff
@@ -1000,6 +1028,35 @@ okapiHeaders, null, handlePreviousResponse(false, false, false,
     }
   }
 
+  private boolean isLoginLegacy(String endpoint) {
+    return endpoint.equals(LOGIN_ENDPOINT_LEGACY);
+  }
+
+  private javax.ws.rs.core.Response respond201WithCookieHeadersAndTokenExpiration(CompletableFuture<Response> loginResponse, CompositeUser cu)
+      throws InterruptedException, ExecutionException {
+    Response response = loginResponse.get();
+    JsonObject body = response.getBody();
+    String accessTokenExpiration = body.getString("refreshTokenExpiration");
+    String refreshTokenExpiration = body.getString("accessTokenExpiration");
+    var tokenExpiration = new TokenExpiration();
+    tokenExpiration.setAccessTokenExpiration(accessTokenExpiration);
+    tokenExpiration.setRefreshTokenExpiration(refreshTokenExpiration);
+    cu.setTokenExpiration(tokenExpiration);
+
+    List<String> setCookieHeaders = response.getHeaders().getAll("Set-Cookie");
+    var setCookieHeadersArray = setCookieHeaders.toArray();
+
+    // Use the ResponseBuilder rather than RMB-generated code. We need to do this because
+    // RMB generated-code does not allow multiple headers with the same key, which is what we need
+    // here.
+    return javax.ws.rs.core.Response.status(201)
+        .header("Set-Cookie", setCookieHeadersArray[0])
+        .header("Set-Cookie", setCookieHeadersArray[1])
+        .type(MediaType.APPLICATION_JSON)
+        .entity(cu)
+        .build();
+  }
```

Comment:
Given that this change includes a major version shift in RMB. 

Does the presence of this code mean that RMB won't be changed to support multiple cookies and this workaround will need to be permanent?

Reply:
The RAML tool we use simply does not allow multiple headers with same key. See https://issues.folio.org/browse/RMB-927 for details.

Reply:
@adamdickmeiss Thank you for helping me understand better.

@steveellis please amend to comment to make it obvious this is a permanent workaround for as long as mod-users-bl uses RMB.